app/src/main/java/org/wikipedia/history/HistoryFragment.java

```diff
@@ -337,137 +330,169 @@ private void restartLoader() {
 
         @Override
         public void onLoadFinished(@NonNull Loader<Cursor> cursorLoader, Cursor cursor) {
-            adapter.setCursor(cursor);
+            List<Object> list = new ArrayList<>();
+            while (cursor.moveToNext()) {
+                IndexedHistoryEntry indexedEntry = new IndexedHistoryEntry(cursor);
+                // Check the previous item, see if the times differ enough
+                // If they do, display the section header.
+                // Always do it this is the first item.
```

Comment:
Very minor typo... missing "if"

Reply:
I copied it from the old logic, and...done.